### PR TITLE
OHRM5X-668: Improve Password Validation in Backend

### PR DESCRIPTION
**Pull request by devishke-orange**
PR contains:
- OHRM5X-668: OS Internal Security Assessment - Password policy configuration should be implemented in backend

### CODE REVIEW

File: symfony/client/src/orangehrmAdminPlugin/pages/location/EditLocation.vue [hunk 1]
```diff
@@ -21,7 +21,9 @@
 <template>
   <div class="orangehrm-background-container">
     <div class="orangehrm-card-container">
-      <oxd-text class="orangehrm-main-title" tag="h6">{{ $t('admin.edit_location') }}</oxd-text>
+      <oxd-text class="orangehrm-main-title" tag="h6">{{
+        $t('admin.edit_location')
+      }}</oxd-text>
```
Comment (RajithaKumara) [inline symfony/client/src/orangehrmAdminPlugin/pages/location/EditLocation.vue:26]:
```suggestion
      <oxd-text class="orangehrm-main-title" tag="h6">
        {{ $t('admin.edit_location') }}
      </oxd-text>
```

File: symfony/client/src/orangehrmAdminPlugin/pages/location/SaveLocation.vue [hunk 2]
```diff
@@ -21,7 +21,9 @@
 <template>
   <div class="orangehrm-background-container">
     <div class="orangehrm-card-container">
-      <oxd-text tag="h6" class="orangehrm-main-title">{{ $t('admin.add_location') }}</oxd-text>
+      <oxd-text tag="h6" class="orangehrm-main-title">{{
+        $t('admin.add_location')
+      }}</oxd-text>
```
Comment (RajithaKumara) [inline symfony/client/src/orangehrmAdminPlugin/pages/location/SaveLocation.vue:26]: can manually format

File: symfony/plugins/orangehrmCorePlugin/Api/V2/Validator/Rules/Password.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        $lowercaseMatch = preg_match(self::LOWERCASE_REGEX, $input);
+        $numberMatch = preg_match(self::NUMBER_REGEX, $input);
+        $specialCharMatch = preg_match(self::SPECIAL_CHAR_REGEX, $input);
+
+        return (
+            $uppercaseMatch > 0 &&
+            $lowercaseMatch > 0 &&
+            $numberMatch > 0 &&
+            $specialCharMatch > 0 &&
+            strlen($input) >= 8
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmCorePlugin/Api/V2/Validator/Rules/Password.php:41]: use text helper to get the length. it consider multibyte characters

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: symfony/client/src/orangehrmAdminPlugin/pages/location/EditLocation.vue (hunk 1 above)
Comment (devishke-orange) [reply symfony/client/src/orangehrmAdminPlugin/pages/location/EditLocation.vue]: done

File: symfony/client/src/orangehrmAdminPlugin/pages/location/SaveLocation.vue (hunk 2 above)
Comment (devishke-orange) [reply symfony/client/src/orangehrmAdminPlugin/pages/location/SaveLocation.vue]: changed

File: symfony/plugins/orangehrmCorePlugin/Api/V2/Validator/Rules/Password.php (hunk 3 above)
Comment (devishke-orange) [reply symfony/plugins/orangehrmCorePlugin/Api/V2/Validator/Rules/Password.php]: changed

**Review by RajithaKumara — APPROVED** [review]

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**